src/main/java/in/org/projecteka/hdaf/link/link/Link.java

```diff
@@ -25,19 +28,23 @@ public Link(HIPClient hipClient, ClientRegistryClient clientRegistryClient) {
         //providerid to be fetched from DB using transactionID
         String providerId = "10000005";
 
+        Patient patientInHIP = toHIPPatient(patientId, patientLinkReferenceRequest);
+        in.org.projecteka.hdaf.link.link.model.hip.PatientLinkReferenceRequest patientLinkReferenceRequestHIP =
+                new in.org.projecteka.hdaf.link.link.model.hip.PatientLinkReferenceRequest(
+                        patientLinkReferenceRequest.getTransactionId(), patientInHIP);
         return providerUrl(providerId)
-                .flatMap(s -> s.map(url -> hipClient.linkPatientCareContext(patientId, patientLinkReferenceRequest, url))
+                .flatMap(s -> s.map(url -> hipClient.linkPatientCareContext(patientLinkReferenceRequestHIP, url))
                         .orElse(Mono.error(new Throwable("Invalid HIP")))
                 );
     }
 
-    public Mono<PatientLinkResponse> verifyToken(String patientId, String linkRefNumber, PatientLinkRequest patientLinkRequest) {
+    public Mono<PatientLinkResponse> verifyToken(String linkRefNumber, PatientLinkRequest patientLinkRequest) {
         //from linkRefNumber get TransactionId
         //from transactionID get providerID
         String providerId = "10000005";
         //Check otp for expiry
         return providerUrl(providerId)
-                .flatMap(s -> s.map(url -> hipClient.validateToken(patientId, linkRefNumber, patientLinkRequest, url))
+                .flatMap(s -> s.map(url -> hipClient.validateToken(linkRefNumber, patientLinkRequest, url))
                         .orElse(Mono.error(new Throwable("Invalid HIP")))
```

Comment:
Mono.error(new Throwable("Invalid HIP"), I think we can't say "Invalid HIP" as we are getting the HIP from linkRefNumber?